### DISCUSSION TITLE
Feature Request: ability to set http headers?

### DISCUSSION THREAD

**Discussion by cboettig** [duckdb/duckdb#8344] [Ideas]
The ability of the `duckdb` httpfs extension to query arbitrary http addresses and leverage range-requests without downloading an entire asset really awesome and frankly game-changing.  However, access to some destinations requires custom headers.

Specifically, I'd like to be able to pass the following header for accessing data on on GHCR.  e.g. passing the header on the command line, we can download this parquet file first and then read with duckdb:

```
curl -fL --header "Authorization: Bearer QQ==" https://ghcr.io/v2/cboettig/content-store/blobs/sha256:871017a6573cf4cd1ae303644de6bd0b04a1029278bfc80c94c37f968687e6d4 -o test.parquet
```

but because the header is required, we cannot use that URL directly in a call to `parquet_scan()` in duckdb.  

(Note that container registries are rapidly being re-imagined as more general asset repositories, a la https://oras.land.  The above header is required for access of public assets on GHCR.)

I imagine there are other cases where setting the http headers would also be useful of course.  Maybe these could be configured with an additional `http_headers` argument to the existing [configuration](https://duckdb.org/docs/sql/configuration.html)?  Or maybe better that this is passed as an argument to `parquet_scan` etc?

**Comment by nicosuave** [comment]
There's a prior discussion [here](https://github.com/duckdb/duckdb/discussions/6782) FYI if you could upvote that one

Edit: also check out a more active discussion and solution (assuming you're using Python [here](https://github.com/duckdb/duckdb/discussions/4963#discussioncomment-6068626)

**Reply by cboettig** [reply]
Thanks @nicosuave, done !  

Cool work-around with `fsspec` for python users!   I have a workflow that needs to support both R and python users, for which duckdb is just perfect in general, but thus would love to see this addressed in duckdb's httpfs.  (FWIW, we've also seen substantially better performance in python using the native duckdb httpfs than when routing via fsspec).

**Reply by Mause** [reply]
Performance is something we explicitly call out in the DuckDB docs for fsspec yeah, as it's definitely not expected to match the httpfs performance
